crates/iroha_executor/src/default/mod.rs

```diff
@@ -1235,40 +1231,49 @@ pub mod role {
         isi: &Register<Role>,
     ) {
         let role = isi.object();
+        let grant_role = &Grant::account_role(role.id().clone(), role.grant_to().clone());
         let mut new_role = Role::new(role.id().clone(), role.grant_to().clone());
 
         // Exception for multisig roles
-        let mut is_multisig_role = false;
-        if let Some(tail) = role
-            .id()
-            .name()
-            .as_ref()
-            .strip_prefix("multisig_signatory_")
         {
-            let Ok(account_id) = tail.replacen('_', "@", 1).parse::<AccountId>() else {
-                deny!(executor, "Violates multisig role format")
-            };
-            if crate::permission::domain::is_domain_owner(
-                account_id.domain(),
-                &executor.context().authority,
-                executor.host(),
-            )
-            .unwrap_or_default()
-            {
-                // Bind this role to this permission here, regardless of the given contains
-                let permission = CanExecuteTrigger {
-                    trigger: format!(
-                        "multisig_transactions_{}_{}",
-                        account_id.signatory(),
-                        account_id.domain()
+            use crate::permission::domain::is_domain_owner;
+
+            const DELIMITER: char = '/';
+            const MULTISIG_SIGNATORY: &str = "MULTISIG_SIGNATORY";
+
+            fn multisig_account_from(role: &RoleId) -> Option<AccountId> {
+                role.name()
+                    .as_ref()
+                    .strip_prefix(MULTISIG_SIGNATORY)?
+                    .rsplit_once(DELIMITER)
+                    .and_then(|(init, last)| {
+                        format!("{last}@{}", init.trim_matches(DELIMITER))
+                            .parse()
+                            .ok()
+                    })
+            }
+
+            if role.id().name().as_ref().starts_with(MULTISIG_SIGNATORY) {
+                if let Some(multisig_account) = multisig_account_from(role.id()) {
+                    if is_domain_owner(
+                        multisig_account.domain(),
+                        &executor.context().authority,
+                        executor.host(),
                     )
-                    .parse()
-                    .unwrap(),
-                };
-                new_role = new_role.add_permission(permission);
-                is_multisig_role = true;
-            } else {
-                deny!(executor, "Can't register multisig role")
+                    .unwrap_or_default()
+                    {
+                        let isi = &Register::role(new_role);
+                        if let Err(err) = executor.host().submit(isi) {
+                            deny!(executor, err);
+                        }
+                        execute!(executor, grant_role);
```

Comment:
```suggestion
                       executor.host().submit(grant_role);
```

I think we should use `Iroha::submit` around this file whenever executing an instruction that is not the instruction being visited. Not completely sure

Reply:
Is your suggestion to give these syntaxes a semantic difference?
If so, seems not so solid practice. What do you think about e.g. wrapping a not yet validated instruction to unwrap on validation to execute, or to execute the wrapped instruction as another method bypassing validation

Reply:
oh, I think I was wrong. Any instruction that is executed from executor, whether via `execute!` or `host.submit` does not undergo any validation. There is no point in switching to `host.submit` you have to make sure to always call `visit_xxx` instead of `execute!` or `host.submit`

Reply:
so you might want to use `visit_grant_role` here instead

Reply:
No, this change 2522e18eaadb95069f9319e9e9b6ae18413d4c99 causes:

```Operation is not permitted: Can't grant or revoke role to another account```

We should skip validation here, because no one at first has permission to grant the newly registered role, which should have been a concept of `NewRole.grant_to`